### PR TITLE
Fix ci failed tests

### PR DESCRIPTION
**Pull request by Nabil-Salah**
### Description

fix ci fails on zos4 

### Changes

`pkg/makefile`  changed testing privilage

`./storage/filesystem/btrfs_ci_test.go` , `pkg/netlight/public/public_test.go` , `pkg/netlight/bridge/bridge_test.go` files have slightly changed usually was setup problems

note: in `./storage/filesystem/btrfs_ci_test.go` i couldn't find the overhead usage of 16384 bytes to test

### Checklist

- [x] Tests included
- [x] Build pass
- [ ] Documentation
- [x] Code format and docstring

### CODE REVIEW

File: pkg/netlight/public/public_test.go [hunk 1]
```diff
@@ -27,6 +28,12 @@ func TestCreatePublicNS(t *testing.T) {
...
 	}()
 
-	err := setupPublicNS(pkg.StrIdentifier(""), iface)
+	_, err := bridge.New("br-pub")
+	require.NoError(t, err)
+	defer func() {
+		err = bridge.Delete("br-pub")
+		require.NoError(t, err)
+	}()
+	err = setupPublicNS(pkg.StrIdentifier(""), iface)
```
Comment (Eslam-Nawara) [inline pkg/netlight/public/public_test.go:37]: you can use `ensurePublicBridge` instead, please don't use `pr-pub` if needed use the `PublicBridge` exported const
https://github.com/threefoldtech/zos4/blob/52bec593eae75cffcd1b9eae60989b40713be1b3/pkg/netlight/types/constant.go#L15
https://github.com/threefoldtech/zos4/blob/52bec593eae75cffcd1b9eae60989b40713be1b3/pkg/netlight/public/public.go#L46

File: pkg/netlight/bridge/bridge_test.go [hunk 2]
```diff
@@ -88,14 +89,19 @@ func TestAttachBridgeWithMac(t *testing.T) {
 	}()
 
 	addrBefore := br.Attrs().HardwareAddr
+	t.Logf("initial bridge mac: %s", addrBefore)
 	err = AttachNicWithMac(dummy, br)
 	require.NoError(t, err)
+	time.Sleep(100 * time.Millisecond)
```
Comment (Eslam-Nawara) [inline pkg/netlight/bridge/bridge_test.go:95]: is that needed?

File: pkg/netlight/bridge/bridge_test.go [hunk 3]
```diff
@@ -88,14 +89,19 @@ func TestAttachBridgeWithMac(t *testing.T) {
...
 
 	l, err := netlink.LinkByName("bro0")
 	require.NoError(t, err)
 	br = l.(*netlink.Bridge)
 	addrAfter := br.Attrs().HardwareAddr
 
+	t.Logf("bridge mac After: %s", addrAfter)
+	t.Logf("dummy mac: %s", dummy.Attrs().HardwareAddr)
```
Comment (Eslam-Nawara) [inline pkg/netlight/bridge/bridge_test.go:103]: remove unnecessary logs please

File: pkg/Makefile [hunk 4]
```diff
@@ -19,7 +20,7 @@ test: lint build
 	go test -vet=off -v $(shell go list ./... | grep -Ev "stubs|network" )
 
 testrace: lint build
-	go test -vet=off -v $(shell go list ./... | grep -Ev "stubs|network" )
+	sudo -E ${GOBINARY} test -vet=off -v $(shell go list ./... | grep -Ev "stubs|network" )
```
Comment (Eslam-Nawara) [inline pkg/Makefile:23]: is it needed to run lint with sudo?

**Comment by Eslam-Nawara** [comment]
I see the linter here was complaining about unused variables, is that fixed? https://github.com/threefoldtech/zos4/actions/runs/12140161674/job/33849323103

File: pkg/storage/filesystem/btrfs_ci_test.go [hunk 5]
```diff
@@ -38,11 +38,17 @@ func (d TestDevices) Loops() Devices {
 	mgr := lsblkDeviceManager{
 		executer: executerFunc(run),
 	}
-	for _, loop := range d {
+	for loopTempPath, loop := range d {
+		size, err := FilesUsage(loopTempPath)
+		if err != nil {
+			panic(err)
+		}
```
Comment (Eslam-Nawara) [inline pkg/storage/filesystem/btrfs_ci_test.go:45]: require error not nil please, don't panic

**Comment by Eslam-Nawara** [comment]
I think your're not using the latest on main branch you're behind few commits (18)

> I see the linter here was complaining about unused variables, is that fixed? https://github.com/threefoldtech/zos4/actions/runs/12140161674/job/33849323103

File: pkg/Makefile (hunk 4 above)
Comment (Nabil-Salah) [reply pkg/Makefile:23]: this sudo is for tests, tests need to run in sudo because of netlink package

Comment (Eslam-Nawara) [reply pkg/Makefile:23]: I think that is only on our local machines, github workflows shouldn't have that issue, if anyone need to run the test they can run it with sudo

File: pkg/storage/filesystem/btrfs_ci_test.go [hunk 6]
```diff
@@ -34,20 +34,26 @@ var (
...
+			Size: size,
 		})
+
 	}
 
 	devices, err := mgr.Devices(context.Background())
 	if err != nil {
-		panic(err)
+		require.NoError(t, err)
 	}
```
Comment (Eslam-Nawara) [inline pkg/storage/filesystem/btrfs_ci_test.go:57]: you should remove the error check, replace this with only `require.NoError(t, err)`

File: pkg/storage/filesystem/btrfs_ci_test.go (hunk 5 above)
Comment (Eslam-Nawara) [reply pkg/storage/filesystem/btrfs_ci_test.go]: also require no error replaces the if clause you shouldn't add it inside the if body

**Review by Eslam-Nawara — APPROVED** [review]

**Pull request merged**